tests/scripts/all.sh

```diff
@@ -897,26 +883,46 @@ component_test_full_cmake_clang () {
     if_build_succeeded env OPENSSL_CMD="$OPENSSL_NEXT" tests/compat.sh -e '^$' -f 'ARIA\|CHACHA'
 }
 
-component_build_deprecated () {
-    msg "build: make, full config + DEPRECATED_WARNING, gcc -O" # ~ 30s
+component_test_default_no_deprecated () {
+    msg "build: make, default + MBEDTLS_DEPRECATED_REMOVED" # ~ 30s
+    scripts/config.py set MBEDTLS_DEPRECATED_WARNING
+    make CC=gcc CFLAGS='-O -Werror -Wall -Wextra'
```

Comment:
Pre-existing in a sense: `-Wall -Wextra` are redundant here, they're already in `WARNING_CFLAGS` which we're not overriding. (I intend to make a PR with this and other clean-ups at some point, but haven't got around to it yet, and should probably wait until you finished you "run components in sub-shells" work to avoid conflicts.)

Reply:
They're redundant, but harmless. In this case I like having them here, because I find it clearer to have all the warning/error-related options in one place.

Reply:
Sure, but in 2.16 and 2.7 you don't actually have all the warning/error options in one place, as `WARNING_CFLAGS` also has `-Wdeclaration-after-statement` (and also `-Wunused` in the `tests` directory). It seems a bit weird to me to be repeating some of them here but not all, but OTOH I agree this is harmless and whether it's better style to repeat them or not is probably not the most important discussion to have now.